devito/data/allocators.py

```diff
@@ -92,8 +92,12 @@ def initialize(cls):
         return
 
     def alloc(self, shape, dtype, padding=0):
-        datasize = int(reduce(mul, shape))
-        ctype = dtype_to_ctype(dtype)
+        # For complex number, allocate double the size of its real/imaginary part
```

Comment:
potentially useful elsewhere, so I'd move it into a function inside `devito/tools/dtypes_lowering` maybe?